UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/Helpers/EssEndpointHelper.cs

```diff
@@ -46,12 +44,28 @@ public static async Task<HttpResponseMessage> ProductVersionsEndpoint(string? ba
 
             string payloadJson = JsonConvert.SerializeObject(productVersion);
 
-            using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, uri)
+            using var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, uri)
+                { Content = new StringContent(payloadJson, Encoding.UTF8, "application/json") };
+            return await httpClient.SendAsync(httpRequestMessage, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// This Method is used to check the Batch Status and Download ES
+        /// </summary>
+        /// <param name="batchId">Provide the BatchId of the requested Product</param>
+        /// <returns></returns>
+        public static async Task<string> CreateExchangeSetFile(string batchId)
+        {
+            // The below sleep is to give time to BuilderService to download the required zip
+            Thread.Sleep(90000);
```

Comment:
Is this in the required spot? It doesn't make sense to sleep the thread right at the start of the method?

Reply:
Thanks Dan, We have moved that to a new method.